zipline/data/us_equity_loader.py

```diff
@@ -0,0 +1,230 @@
+        return self.current
+
+
+class USEquityHistoryLoader(object):
+    """
+    Loader for sliding history windows of adjusted US Equity Pricing data.
+
+    Parameters
+    ----------
+    daily_reader : DailyBarReader
+        Reader for daily bars.
+    adjustement_reader : SQLiteAdjustmentReader
+        Reader for adjustment data.
+    """
+
+    def __init__(self, daily_reader, adjustment_reader):
+        self._daily_reader = daily_reader
+        self._calendar = daily_reader._calendar
+        self._adjustments_reader = adjustment_reader
+        self._daily_window_blocks = {}
+
+        self._prefetch_length = 40
+
+    def _get_adjustments_in_range(self, assets, days, field):
+        """
+        Get the Float64Multiply objects to pass to an AdjustedArrayWindow.
+
+        For the use of AdjustedAraryWindow in the loader, which looks back
+        from current simulation time back to a window of data the dictionary is
+        structured with:
+        - the key into the dictionary for adjustments is the location of the
+        day from which the window is being viewed.
+        - the start of all multiply objects is always 0 (in each window all
+          adjustments are overlapping)
+        - the end of the multiply object is the location before the calendar
+          location of the adjustment action, making all days before the event
+          adjusted.
+
+        Parameters
+        ----------
+        assets : iterable of Asset
+            The assets for which to get adjustments.
+
+        days : iterable of datetime64-like
+            The days for which adjustment data is needed.
+        field : str
+            OHLCV field for which to get the adjustments.
+
+        Returns
+        -------
+        out : The adjustments as a dict of loc -> Float64Multiply
+        """
+        sids = {int(asset): i for i, asset in enumerate(assets)}
+        start = days[0]
+        end = days[-1]
+        adjs = {}
+        for sid, i in iteritems(sids):
+            if field != 'volume':
+                mergers = self._adjustments_reader.get_adjustments_for_sid(
+                    'mergers', sid)
+                for m in mergers:
+                    dt = m[0]
+                    if start < dt <= end:
+                        end_loc = days.get_loc(dt)
+                        adjs[end_loc] = [Float64Multiply(0,
+                                                         end_loc - 1,
+                                                         i,
+                                                         i,
+                                                         m[1])]
+                divs = self._adjustments_reader.get_adjustments_for_sid(
+                    'dividends', sid)
+                for d in divs:
+                    dt = d[0]
+                    if start < dt <= end:
+                        end_loc = days.get_loc(dt)
+                        adjs[end_loc] = [Float64Multiply(0,
```

Comment:
I think this is broken if we have a merger and a dividend or a split on the same date? (We're just overwriting it seems like?)